doc/api/cli.md

```diff
@@ -288,6 +288,17 @@ Path to the file used to store the persistent REPL history. The default path is
 to an empty string (`""` or `" "`) disables persistent REPL history.
 
 
+### `NODE_TTY_UNSAFE_ASYNC=1`
+<!-- YAML
+added: REPLACEME
+-->
+
+When set to `1` writes stdout and stderr will be non-blocking and async when
+outputting to a TTY on platforms which support async stdio.
```

Comment:
> Do we have an accurate understanding of Windows

Erm, maybe.

I'm 99% sure Windows is always async to a Terminal because windows doesn't give you a way to tell if a write ever actually made it to the terminal. (Sources: Bert and Trevor, I think.)

Pipes definitely block on windows. (Source: [this code in `net.js`](https://github.com/nodejs/node/blob/c09dfe3a0bf63ecff522a7f9c06c21593969e69d/lib/net.js#L140-L144))

I have no idea about files & windows.